RecoMET/METFilters/plugins/ChargedHadronTrackResolutionFilter.cc

```diff
@@ -0,0 +1,149 @@
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/METReco/interface/PFMET.h"
+#include "DataFormats/METReco/interface/PFMETFwd.h"
+#include "DataFormats/METReco/interface/PFMETCollection.h"
+//
+// class declaration
+//
+
+class ChargedHadronTrackResolutionFilter : public edm::EDFilter {
+public:
+  explicit ChargedHadronTrackResolutionFilter(const edm::ParameterSet&);
+  ~ChargedHadronTrackResolutionFilter();
+
+private:
+  virtual bool filter(edm::Event&, const edm::EventSetup&) override;
+
+      // ----------member data ---------------------------
+
+  edm::EDGetTokenT<reco::PFCandidateCollection>   tokenPFCandidates_;
+  edm::EDGetTokenT<reco::PFMETCollection>   tokenPFMET_;
+  const bool taggingMode_;
+  const double          ptMin_;
+  const double          metSignifMin_;
+  const bool debug_;
+
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+ChargedHadronTrackResolutionFilter::ChargedHadronTrackResolutionFilter(const edm::ParameterSet& iConfig)
+  : tokenPFCandidates_ ( consumes<reco::PFCandidateCollection>(iConfig.getParameter<edm::InputTag> ("PFCandidates")  ))
+  , tokenPFMET_ (        consumes<reco::PFMETCollection>(iConfig.getParameter<edm::InputTag> ("PFMET")  )) 
+  , taggingMode_          ( iConfig.getParameter<bool>    ("taggingMode")         )
+  , ptMin_                ( iConfig.getParameter<double>        ("ptMin")         )
+  , metSignifMin_         ( iConfig.getParameter<double>        ("MetSignifMin")  )
+  , debug_                ( iConfig.getParameter<bool>          ("debug")         )
+{
+  produces<bool>();
+}
+
+ChargedHadronTrackResolutionFilter::~ChargedHadronTrackResolutionFilter() { }
+
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+bool
+ChargedHadronTrackResolutionFilter::filter(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  using namespace std;
+  using namespace edm;
+
+  Handle<reco::PFCandidateCollection>     pfCandidates;
+  iEvent.getByToken(tokenPFCandidates_,pfCandidates);
+  Handle<reco::PFMETCollection>     pfMET;
+  iEvent.getByToken(tokenPFMET_,pfMET);
+
+  bool foundBadTrack = false;
+  if ( debug_ ) cout << "starting loop over pfCandidates" << endl;
+
+  for ( unsigned i=0; i<pfCandidates->size(); ++i ) {
+
+    const reco::PFCandidate & cand = (*pfCandidates)[i];
+  
+    if ( fabs(cand.pdgId()) == 211 ) {
+      // if ( debug_ ) cout << "Found charged hadron candidate" << std::endl;
```

Comment:
please cleanup if (debug_) commented out lines or recode with edm::LogTrace if this is really needed for debugging